README


This archive contains the distribution MooseX-Getopt,
version 0.67:

  A Moose role for processing command line options

This software is copyright (c) 2007 by Infinity Interactive, Inc.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.032.



Changes
Revision history for MooseX-Getopt

0.67      2015-02-28 04:37:19Z
 - remove argv from passed-through buildargs if no argv attribute exists, to
   mollify MooseX::StrictConstructor (RT#101938, Jose Luis Martinez)

0.66      2015-01-28 02:05:54Z
 - fix tests for formatting changes in Getopt::Long::Descriptive 0.099
   (RT#101724)

0.65      2014-08-06 06:09:04Z
 - re-release to remove README.pod from shipped dist

0.64      2014-08-02 17:35:20Z
 - bump required version of Getopt::Long::Descriptive to avoid an infinite
   loop problem (thanks, Bram!)

0.63      2014-03-25 16:37:58Z
 - fix test prereq on Test::Warnings (thanks, omega!)

0.62      2014-03-07 05:44:01Z
 - fix test that assumes option parsing is case-insensitive, which is no
   longer true with Getopt::Long::Descriptive 0.097

0.61      2014-02-11 06:30:14Z
 - fix potentially-broken check-breaks test when a conflicting module fails to
   compile

0.60      2014-02-10 02:21:05Z
 - avoid failing an optional test when YAML is not installed

0.59      2013-11-30 19:02:01Z
 - re-release to fix compile test

0.58      2013-09-13 16:57:41Z
 - remove use of deprecated $Config::Any::YAML::NO_YAML_XS_WARNING in tests
 - warnings tests bypassed during installation
 - author metadata moved to contributors

0.57      2013-09-07 21:29:05Z
 - spelling fixes (RT#87780, David Steinbrunner)
 - removed use of deprecated Class::MOP::load_class
 - repository migrated from shadowcat to the github moose organization

0.56      2013-02-15 21:13:18Z
 - depend on newest Path::Tiny, to regain 5.8.x compatibility (gone since
   v0.52)

0.55      2013-02-12 18:02:55Z
 - fix broken tests on win32 with file comparisons

0.54      2013-02-10 00:25:32Z
 - fix broken tests on win32 with file comparisons
 - allow configfiles called "0"
 - support more mechanisms for overriding default configfile
 - when using in conjunction with MooseX::ConfigFromFile, configfile value now
   properly passed to new()

0.53      2013-02-05 17:59:00Z
 - properly indicate optional dependency in tests using
   MooseX::ConfigFromFile

0.52      2013-02-03 03:01:25Z
 - tests converted from using Path::Class to Path::Tiny

0.51      2013-01-27 17:38:11Z
 - missing test prereqs now declared (thanks for the automated reports, David
   Cantrell!)

0.50      2012-12-27 08:20:56Z
 - unbreak Catalyst::Runtime tests (etc) by keeping _getopt_full_usage in the
   callstack. (RT#82249)

0.49      2012-12-26 18:13:41Z
 - silence a warning in tests if the user does not have YAML::XS installed
   (thanks for the report, Getty!)
 - fix off-by-one issues in test plans if optional requirements are not met

0.48      2012-12-24 02:08:04Z
 - documentation and tests amended to prefer usage of 'trait' over 'metaclass'
   as much as possible
 - print_usage_text() exposed as a public method, to make it easier for
   consuming classes to modify the behaviour when usage text is printed

0.47      2012-08-30 23:37:59Z
 - re-release to remove double $VERSION declarations

0.46      2012-08-30 22:17:28Z
 - -h is also treated as a --help option

0.45      2012-05-13 16:49:58Z
 - use Path::Class in test to avoid test failures on MSWin32 with filename
   construction

0.44      2012-05-11 17:44:48Z
 - now respecting value of configfile passed to the constructor, when used in
   conjunction with MooseX::ConfigFromFile. (Karen Etheridge)

0.43      2012-04-29 16:35:52Z
 - re-release as non-trial.

0.42      2012-04-15 16:13:05Z (TRIAL RELEASE)
 - reformatted the Changes file, with the help of CPAN::Changes, to something
   a little more standard.
 - more release-only tests added

0.41      2012-04-15 01:36:00Z (TRIAL RELEASE)
 - Now bails with exit status 0, rather than dying, when printing
   requested --usage/--help information (usage printed via invalid arguments
   will still die, however). (Karen Etheridge)

0.40      2012-04-13
 - Fix tests when MooseX::ConfigFromFile is installed, with
   Getopt::Long::Descriptive >= 0.091. RT#76287
 - Pod fix from debian. Salvatore Bonaccorso <carnil@debian.org> RT#74681
 - Spelling fix from debian. Jonathan Yu <jawnsy@cpan.org> RT#74681

0.39      2012-02-03
 - Fix argument parsing in process_argv when used with ConfigFromFile so
   that -v can be used as expected, rather than being grabbed as --version

0.38      2011-12-23
 - More documentation on Getopt::Long settings and how they alter this
   role's behaviour, in particular with regards to extra_argv capturing.
   (Karen Etheridge)

0.37      2011-03-30
 - Fix missing dependency. (RT#67077)

0.36      2011-03-29
 - Fix dist.ini 'Prereq' into 'Prereqs' (Damien Krotkine)
 - MooseX::Getopt::GLD
   - change it to a MooseX::Role::Parameterized, so that it accepts
     'getopt_conf' parameter (Damien Krotkine)

0.35      2011-02-09
 - Fix missed change from Test::Exception to Test::Fatal

0.34      2011-02-08
 - The test suite now uses Test::Fatal instead of Test::Exception. (Karen
   Etheridge)
 - Move most of new_with_options logic into separate publically accessible
   process_argv method, so that arg processing can be done without actually
   creating object (Jonathan Swartz)

0.33      2010-08-26
 - Remove Test::Most from newly added test (RT#60766)

0.32      2010-08-26
 - Fix handling of Union types (RT#58417)

0.31      2010-07-07
 - Fix issue causing tests to fail if MooseX::SimpleConfig isn't installed.

0.30      2010-07-06
 - MooseX::Getopt::Basic
   - uses attribute insertion order when determining the order in which
     options are documented in the usage (e.g. in $obj->usage->text). (Karen
     Etheridge)

 - MooseX::Getopt::Basic
   - store the usage object to the usage attr (RT#58715)
   - properly checks whether the *option* --help, --usage, or --? were used,
     rather than the attribute 'help', 'usage' or '?' were set
 - MooseX::Getopt::GLD
   - add the usage attribute for storing the Getopt::Long::Descriptive::Usage
     object, and the help attribute for storing --help, --usage and --? state

Makefile.PL
# This Makefile.PL for MooseX-Getopt was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.33.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "A Moose role for processing command line options",
  "AUTHOR" => "Stevan Little <stevan\@iinteractive.com>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "MooseX-Getopt",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "MooseX::Getopt",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Getopt::Long" => "2.37",
    "Getopt::Long::Descriptive" => "0.088",
    "Moose" => 0,
    "Moose::Meta::Attribute" => 0,
    "Moose::Role" => "0.56",
    "Moose::Util::TypeConstraints" => 0,
    "MooseX::Role::Parameterized" => 0,
    "Scalar::Util" => 0,
    "Try::Tiny" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Module::Runtime" => 0,
    "Moose::Meta::Class" => 0,
    "Moose::Util" => 0,
    "Path::Tiny" => "0.009",
    "Test::Deep" => 0,
    "Test::Fatal" => "0.003",
    "Test::Moose" => 0,
    "Test::More" => "0.88",
    "Test::Requires" => 0,
    "Test::Trap" => 0,
    "Test::Warnings" => "0.009",
    "if" => 0
  },
  "VERSION" => "0.67",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Carp" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Getopt::Long" => "2.37",
  "Getopt::Long::Descriptive" => "0.088",
  "Module::Build::Tiny" => "0.039",
  "Module::Runtime" => 0,
  "Moose" => 0,
  "Moose::Meta::Attribute" => 0,
  "Moose::Meta::Class" => 0,
  "Moose::Role" => "0.56",
  "Moose::Util" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "MooseX::Role::Parameterized" => 0,
  "Path::Tiny" => "0.009",
  "Scalar::Util" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => "0.003",
  "Test::Moose" => 0,
  "Test::More" => "0.88",
  "Test::Requires" => 0,
  "Test::Trap" => 0,
  "Test::Warnings" => "0.009",
  "Try::Tiny" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
